### PR TITLE
Add explicit imports across files inside of polymer

### PR DESCRIPTION
**Pull request by FredKSchott**
This PR adds back missing imports. These explicit references are needed for some automated tooling.

/cc @justinfagnani

### CODE REVIEW

File: lib/utils/style-gather.html [hunk 1]
```diff
@@ -8,6 +8,7 @@
 subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
 -->
 <link rel="import" href="resolve-url.html">
+<link rel="import" href="../elements/dom-module.html">
```
Comment (justinfagnani) [inline lib/utils/style-gather.html]: This file shouldn't have an import of dom-module. The check below is to see if it's loaded and use it if so, but there are some uses of style-gather that don't necessitate loading dom-module, like custom-style.

Comment (FredKSchott) [inline lib/utils/style-gather.html]: Agreed, removing

**Comment by FredKSchott** [comment]
@justinfagnani @sorvell PTAL, PR is now a single line

**Comment by justinfagnani** [comment]
LGTM from my end

**Pull request merged**